docs/en/operation/User-CLI.md

```diff
@@ -1302,6 +1302,33 @@ $ ./bin/alluxio fs rm /tmp/unused-file
 $ ./bin/alluxio fs rm --alluxioOnly /tmp/unused-file2
 ```
 
+When deleting only from Alluxio but leaving the files in UFS, you can use `-U` and `-Dalluxio.user.file.metadata.sync.interval=-1`
+to skip the metadata sync and the UFS check. This will save time and memory consumption on the Alluxio master.
+```console
+$ bin/alluxio fs rm -R -U --alluxioOnly -Dalluxio.user.file.metadata.sync.interval=-1 /path
+```
+
+When deleting a large directory (with millions of files) recursively both from Alluxio and UFS,
+the operation is expensive. 
+
+We recommend doing the deletion in the following way:
+1. Using the corresponding tool to check the UFS path directly and make sure everything can be deleted safely. 
+For example if the UFS is HDFS, use `hdfs dfs -ls -R /dir` to list the UFS files and check.
+Please do not sync the directory from Alluxio and check with `alluxio fs ls -R /dir`, because the loaded file metadata will
+be deleted anyway and the expensive metadata sync operation will essentially be wasted.
+2. Then issue the deletion from Alluxio and delete both in-Alluxio and in-UFS files:
+```console
+# Disable the sync and skip the UFS check, to reduce memory consumption on the master side
+$ bin/alluxio fs rm -R -U -Dalluxio.user.file.metadata.sync.interval=-1 /dir
+```
+
+If the metadata sync and UFS check are both disabled, deleting 1 million files from Alluxio will increase the JVM heap
```

Comment:
this sounds too much like a story. let's just lay down the facts:
using an example of deleting 1 million files:
- if both metadata sync and ufs check are disabled, the jvm heap will increase by 2.5GB
- if only ufs check is disabled (but metadata syns is enabled?), the jvm heap will increases by 2.7GB (from 2.5GB +10%?)
- if both metadata sync and ufs check are enabled, the jvm heap will increase by 4.5 GB (2.5+2? i'm guessing because it says an extra 2 and 2 < 2.5, so i'm assuming i should add them together)

Reply:
When I was writing this part, my intention is to only recommend these 2 use cases:
1. Disable sync and check when `--alluxioOnly`
2. Disable sync and check when deleting UFS too

I tried to avoid mentioning too much about the other options because users should not use those and better not ask. And we are not too interested in giving a precise estimation on use cases we don't recommend, nor supporting(updating) those numbers in future versions.

That said, do you prefer my current description in the PR or listing the numbers as your recommendation? Both flavor look fine to me, and your math is correct.

Reply:
okay great, then list those exact cases and the memory overhead for each with 1 million files as the example.

currently if i had to do the math for the second case, i have to take the proportion from the first example and then apply the 10%. and then when am i supposed to apply the 20% reduction mentioned for 2.8; was this accounted for already? now i am a terribly confused reader

Reply:
Ok i updated the section as recommended, see the new numbers, those are all for the latest 2.8 benchmark.